src/cloudant/design_document.py

```diff
@@ -79,6 +91,23 @@ def add_view(self, view_name, map_func, reduce_func=None, **kwargs):
         view = View(self, view_name, map_func, reduce_func, **kwargs)
         self.views.__setitem__(view_name, view)
 
+    def add_search_index(self, index_name, search_func, analyzer='standard'):
+        """
+        Appends a Cloudant search index to the locally cached DesignDocument
+        indexes dictionary.
+
+        :param str index_name: Name used to identify the search index.
+        :param str search_func: Javascript search index function.
+        :param analyzer: Optional analyzer for this search index.
+        """
+        if self.get_index(index_name) is not None:
+            msg = ('Search index {0} already exists in this design doc'
```

Comment:
I think this message should be less specific e.g. `An index with name {0} already...` I don't think we have any guarantee that the existing index is a search index. Possibly wouldn't be necessary if a language check happened first and threw a different exception.